subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaDoc.java

```diff
@@ -152,6 +153,17 @@ public void setTitle(@Nullable String title) {
         this.title = title;
     }
 
+    /**
+     * Returns whether it is scala 3.
+     * @since 7.3
+     */
+    @Incubating
+    @Optional
+    @Input
+    public Boolean getIsScala3() {
```

Comment:
Thanks for looking into my suggestion to inject the Scala version down here.
However adding a public field for this feels wrong.

I had a look at the other changes and what could be done instead is reach out to `ScalaBasePlugin.isScala3` method, which would need to be made public.

This would also remove the need for adding a convention mapping - a concept we are trying to move away from.

Reply:
Actually, reviewing #18347, let's just add the `ScalaRuntime` as an `@Internal` field instead.
See https://github.com/gradle/gradle/pull/18347/files#diff-bd022006e2ee4c0e9471eff5abec3f44a6e40a84e760796df120df9818c9ca3cR51